elasticsearch/examples/secure-elasticsearch/templates/jobs.yaml

```diff
@@ -0,0 +1,130 @@
+{{- /* Try to only deploy maintenance/setup jobs when the role is master.
+     * This keeps us from trying to deploy the same job twice if we have
+     * separate master and data nodes defined */ -}}
+{{- if (eq .Values.elasticsearch.roles.master "true") }}
+---
+# This job deploys the Elastic license payload to the cluster once it has formed.
+# Job: {{ .Values.elasticsearch.roles.master }}
+apiVersion: batch/v1
+kind: Job
```

Comment:
I'm trying to imagine how this is supposed to look like. The `command` field in a job spec is immutable so it can't be updated or changed. If you did want to have some kind of "run the job when it changes or gets updated" it would need to do a trick like adding the checksum of the file into the metadata.name to force a new resource to be created.  I think helm would also then cleanup the previous job but I'm not too sure.

Reply:
In practice, helm does poorly with Jobs because it attempts to push a new job config (if the job changes) and is rejected by kubernetes. The only solution is to delete the job, then `helm install` (or upgrade) again. It's annoying.

I think a Job is wrong for this, but I can't think of any other more appropriate solution.

Reply:
Thinking about it more, this is probably best done by a controller or operator-style service which watches the configmap(s) and ships new settings to Elasticsearch when necessary. Such a thing may be out of scope for this helm-chart, though I would use it in production. Alternately, instead of a job, maybe a pod definition that is basically `while true; do sync configmap to elasticsearch if it changes; done` ?